client/extensions/woocommerce/state/ui/shipping/classes/selectors.js

```diff
@@ -0,0 +1,133 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { get, map, find } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getShippingClassOptions } from 'woocommerce/state/sites/shipping-classes/selectors';
+
+/**
+ * A shortcut that extracts the shipping classes UI state from the whole state.
+ *
+ * @param  {Object} state   The current Redux state.
+ * @param  {number} siteId  Site ID.
+ * @return {Object}         The local state.
+ */
+export const getUiShippingClassesState = ( state, siteId = getSelectedSiteId( state ) ) => {
+	return get( state, [ 'extensions', 'woocommerce', 'ui', 'shipping', siteId, 'classes' ], {} );
+};
+
+/**
+ * Loads the list of shipping classes for the UI, which includes:
+ * 1. All saved shipping classes
+ * 2. All newly created shipping classes
+ * 3. All applied updates
+ * 4. But not deleted classes
+ *
+ * This selector loads the interim state of data, as it is manipulated before being saved.
+ *
+ * @param  {Object} state   The current Redux state.
+ * @param  {number} siteId  Site ID.
+ * @return {Array}          All shipping classes.
+ */
+export const getUiShippingClasses = ( state, siteId = getSelectedSiteId( state ) ) => {
+	const localState = getUiShippingClassesState( state, siteId );
+
+	const deleted = get( localState, 'deleted', [] );
+	const updates = get( localState, 'updates', [] );
+	const created = get( localState, 'created', [] );
+
+	// Merge the existing classes and add the basics of newly added ones
+	const merged = [
+		...getShippingClassOptions( state, siteId ),
+
+		// Start with a simple ID, updates will be applied later
+		...map( created, id => ( { id, isNew: true } ) ),
```

Comment:
The `isNew` is no longer used anywhere, is that right?

Reply:
`isNew` is used within the dialog which does not show the "Delete this class" button if the class is new/just being created.

![screenshot 2018-09-11 at 01 07 12](https://user-images.githubusercontent.com/5311119/45329093-09754100-b55f-11e8-984a-680a40e2ad8f.png)

That particular one however was only used to show a placeholder for the slug and is now redundant, so I removed it.